### PR TITLE
[MHLO] fold mhlo.clamp

### PR DESCRIPTION
**Pull request by qingyunqu**
add folder of `mhlo.clamp`

### CODE REVIEW

File: tensorflow/compiler/xla/mlir_hlo/tests/Dialect/mhlo/canonicalize/canonicalize.mlir [hunk 1]
```diff
@@ -252,6 +252,28 @@ func.func @min_fold_float() -> tensor<6xf32> {
   func.return %2 : tensor<6xf32>
 }
 
+// CHECK-LABEL: clamp_scalar_fold
```
Comment (sherhut) [inline tensorflow/compiler/xla/mlir_hlo/tests/Dialect/mhlo/canonicalize/canonicalize.mlir:255]: Could you add a test with float that also covers NaN behavior?

**Review by sherhut — COMMENTED** [review]
Thanks for adding this!

File: tensorflow/compiler/xla/mlir_hlo/tests/Dialect/mhlo/canonicalize/canonicalize.mlir (hunk 1 above)
Comment (qingyunqu) [reply tensorflow/compiler/xla/mlir_hlo/tests/Dialect/mhlo/canonicalize/canonicalize.mlir:255]: ok

Comment (qingyunqu) [reply tensorflow/compiler/xla/mlir_hlo/tests/Dialect/mhlo/canonicalize/canonicalize.mlir:255]: BTW, how to fold if 3.0 < x < 1.0 ? I think the result is NaN?

Comment (sherhut) [reply tensorflow/compiler/xla/mlir_hlo/tests/Dialect/mhlo/canonicalize/canonicalize.mlir:255]: XLA currently would compute min(3.0, max(x, 1.0)), so the folder would do the same thing as code generation does.

**Review by sherhut — APPROVED** [review]

**Pull request merged**